### PR TITLE
[coordinates|TICA] fix dimension() in TICA

### PR DESCRIPTION
**Pull request by fabian-paul**
Fixes dimension issue in #653.
Now the TICA rank supersedes the dim-option of TICA.

``` Python
import pyemma
import numpy as np
x=np.linspace(0,12,1000)
traj = np.zeros((1000,4))
traj[:,0] = np.sin(x)
traj[:,1] = np.sin(x+0.5)
traj[:,2] = np.sin(x+0.4)
traj[:,3] = np.sin(x+0.2)
t = pyemma.coordinates.transform.TICA(1,3)
print t.dimension() # 3
t.fit_transform(traj)
print t.dimension() # 2
```

### CODE REVIEW

**Comment by franknoe** [comment]
looks good, thank you!
